### ISSUE TITLE
Push sugestion when there's no local changes

### ISSUE THREAD

**Issue by antoniodiego** [apache/netbeans#5331]
### Description

I think that will be good if it be showed prompt to make push on a project when there's no local changes a it and there are differences between local anda remote repositories.

Maybe it will help to keeping remote repositories updated avoiding situations like traveling and forgetting to upload changes to the remote repository to continue working on them elsewhere

### Use case/motivation

It should apper a symbol that signals that there are outgoing commit on the project as there's a symbol for singnals local changes

Maybe some other more flashy way will be good, at the show changes tab when the whole project is selected or at the notifications tab whit the push action on it

### Are you willing to submit a pull request?

No

### Code of Conduct

Yes

**Comment by neilcsmith-net** [comment]
Do you have versioning labels enabled?  `Tools / Options / Team / Versioning / Show Versioning Labels`.

**Comment by antoniodiego** [comment]
It was not enabled and I enable it now

I now can see info next to the project names

I think this can already help a little

**Comment by antoniodiego** [comment]
What do you think about leaving it enabled by default ?

**Comment by neilcsmith-net** [comment]
This was discussed recently - cc @mbien 

Good to know it covers what you're looking for.

**Comment by neilcsmith-net** [comment]
I can only see one argument against?!  Versus 3 (now 4) people saying we should.  As with everything (eg. inline hints! :smile:) we need to communicate how to switch it off, but I think this is one of those really useful things that no-one finds.

**Comment by mbien** [comment]
tbh in case of inline hints we should turn it off and communicate how to turn it on again :P

I have a hotkey for it and toggle it on and off on demand - works great.

**Comment by antoniodiego** [comment]
My inspiration to create this issue was the VS Code. 

In it appears a large button for sync local anda remote repositories afeter we do commits and other things that make to have no local changes.

It was so important to me that make I prefer to use VS Code over NetBeans at some moments, as it was remebering to pushing changes at most of the time.

I don't know  if this helps in any way

Pushing acctually has now became an important thing as most of the projects have their codes hosted at remote repositories with teams working on it and some peoples using more than one computer to work on it.

**Comment by Chris2011** [comment]
I don't know whether I get it correctly or not, but pushing to a repo where remote has still new stuff, is not possible by default. It will show you the message, that you need to pull first and then push. Except from push --force.

**Comment by antoniodiego** [comment]
> I don't know whether I get it correctly or not, but pushing to a repo where remote has still new stuff, is not possible by default. It will show you the message, that you need to pull first and then push. Except from push --force.

Yes. You is correct.

**Comment by antoniodiego** [comment]
In the last days I've been noticing that is need very steps to do a push or pull on NetBeans. As example, we ne do a right click, go to Git, after to Remote, Push, next, select the branches, and only then complete the operation.

Maybe this take time and is tiring for some. At least for me it looks tiring.

Today I was thinking if it isn't lacking of a versioning windows on NetBeans. I found the Git repository browsing.

Maybe a push, pull or sync buttons could be put in any of these

**Comment by neilcsmith-net** [comment]
> Maybe a push, pull or sync buttons could be put in any of these

Right-click the main toolbar and add the Git actions.  Does this cover what you're looking for?

**Comment by antoniodiego** [comment]
> I think so

**Comment by antoniodiego** [comment]
> > Maybe a push, pull or sync buttons could be put in any of these
> 
> Right-click the main toolbar and add the Git actions. Does this cover what you're looking for?

 I think so
